airflow/models/dag.py

```diff
@@ -844,11 +844,12 @@ def get_task_instances(
             start_date = timezone.make_aware(
                 datetime.combine(start_date, datetime.min.time()))
 
-        tis = session.query(TaskInstance).filter(
+        tis = session.query(TaskInstance).options(joinedload(TaskInstance.task_tags)).filter(
```

Comment:
Make this an optional flag on this method -- the scheduler doesn't care about tags so shouldn't pay the cost of loading them.

Reply:
I took a look at SQLAlchemy [documentation](https://docs.sqlalchemy.org/en/13/orm/loading_relationships.html) and modified the relationship to raise an exception on lazy loads by default. I then added explicit `joinload`s in a few specific places where we want to load task tags. So currently the tags are not loaded by default and are only loaded with a join in a few places.